File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.go
func findMin(nums []int) int {
	//better solution with O(log n)

	min:=nums[0]
	l,r:=0,len(nums)-1
	for l<=r{
		if nums[l]<nums[r]{
			if nums[l]<min{
				min = nums[l]
			}
			break
		}
		mid:= l+(r-l)/2
		if nums[mid]<min{
			min=nums[mid]
		}
		if nums[mid]>=nums[l]{
			l=mid+1
		}else{
			r=mid-1
		}
	}
	return min

}

File: Data Structures & Algorithms/sort-an-array/submission-4.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        minval,maxval = min(nums), max(nums)
        index = 0
        freq = defaultdict(int)
        for num in nums:
            freq[num] +=1

        for i in range(minval, maxval+1):
            while freq[i]:
                nums[index] = i
                freq[i]-=1
                index +=1

        return nums